### PR TITLE
#208 Upgrade CCM Dependencies, Jenkins, & Jenkins Plugin Versions

### LINKED ISSUE #208: Uplift Jenkins and CCM Dependencies to Latest Versions
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

Newer versions of Jenkins and the .NET dependencies for CCM have been released.

### Describe The Solution. Why is it needed?

Will look to bring all available packages to latest stable and available releases.

### Additional Context

N/A

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by ryanrichter94**
## Description Of Changes
Go through and update the following things:
- Jenkins to V 2.414.1
- dotnet-aspnetcoremodule to V 16.0.23196
- dotnet-6.0-runtime to V 6.0.21
- dotnet-6.0-aspnetruntime to V 6.0.21
- All Jenkins plugins to latest as of 08-25-2023
- Removed old plugin test and added new ones for all current Jenkins plugins we use.

## Motivation and Context
Go through and update Jenkins, Jenkins Plugins, and Chocolatey Central Management dependency package versions to latest available.

## Testing
1. Built on Local VM

### Operating Systems Testing
- Windows Server 2022

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [X] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [X] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [X] Tests to cover my changes, have been added.
* [X] All new and existing tests passed?
* [X] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

Fixes #208

### CODE REVIEW

**Comment by ryanrichter94** [comment]
Still need to go through and remove test for powershell Jenkins plugin and add tests for all other plugins still in list.

**Comment by ryanrichter94** [comment]
Passed my internal testing. Ready for review!

File: Start-C4bJenkinsSetup.ps1 [hunk 1]
```diff
@@ -43,7 +43,7 @@ process {
     [Environment]::SetEnvironmentVariable('JAVA_OPTS', '-Djenkins.install.runSetupWizard=false', 'Machine')
 
     # Install Jenkins
-    $chocoArgs = @('install', 'jenkins', '--version=2.401.2', '-y', "--source='$Source'", '--no-progress')
+    $chocoArgs = @('install', 'jenkins', '--version=2.414.1', '-y', "--source='$Source'", '--no-progress')
```
Comment (steviecoaster) [inline Start-C4bJenkinsSetup.ps1:46]: Why do we not pull latest?

Comment (ryanrichter94) [reply Start-C4bJenkinsSetup.ps1]: As you wish.

**Review by steviecoaster — APPROVED** [review]

**Pull request merged**